runtime/src/prioritization_fee_cache.rs

```diff
@@ -293,6 +293,10 @@ impl PrioritizationFeeCache {
         bank_id: BankId,
         metrics: &PrioritizationFeeCacheMetrics,
     ) {
+        if unfinalized.is_empty() {
+            return;
+        }
```

Comment:
Why does this early return help? Presumably there could be other (newer) slots in the `unfinalized` map when an empty slot is finalized.

Alternate question: are there known times, like on boot, when `finalize_slot` is called but `unfinalized` is empty?

Reply:
1. when a slot is actually empty - no transactions at all, only ticks; the warning will be triggered falsely
2. when a slot has transactions, but none of them set priority fee, so the `unfinalized` cache will be empty, the warning will be triggered falsely too
The early return covers these two cases. 

You brought a good point: "there could be other (newer) slots in the unfinalized map when an empty slot is finalized", it should be a rare case in mnb, do we want to be warned about it? Think about it, probably not necessary - nothing wrong with an slot without priority fee.

Reply:
Okay cool, thanks for walking through that!